code/game/objects/items/sharpbowlerhat.dm

```diff
@@ -0,0 +1,33 @@
+/obj/item/clothing/head/bowler/bowlersharp
+	name = "bowler-hat"
+	desc = "A bowler hat whos edges seem unyielding, hopefully like the gentleman who wears it once you read this."
+	force = 10
+	throwforce = 20
+	throw_speed = 5
+	w_class = WEIGHT_CLASS_SMALL
+	sharpness = IS_SHARP //It can embed to stop too.
+
+/obj/item/clothing/head/bowler/bowlersharp/Initialize()
+	. = ..()
+
+/obj/item/clothing/head/bowler/bowlersharp/throw_at(atom/target, range, speed, mob/thrower, spin=FALSE, diagonals_first = FALSE, datum/callback/callback)
+	unlimitedthrow = TRUE //When you throw it, unlimitedthrow becomes true.
+	. = ..(target, range, speed, thrower, FALSE, diagonals_first, callback)
+
+/obj/item/clothing/head/bowler/bowlersharp/throw_impact(atom/hit_atom)
+	if(!iscarbon(hit_atom))//if the hat hits a thing that isn't carbon.
+		unlimitedthrow = FALSE //Then we mark unlimited throw as false, aka end of the line pal.
+		return ..()
+	var/mob/living/carbon/human/H = hit_atom
+	if(iscarbon(H))
+		H.Knockdown(81) //Heres the downfall of this, they need to go down, or they die to a object boring through them.
```

Comment:
Add a blacklist var to the thrownthing datum for those atoms it should skip on hit_check(), specifically at line 159 throwing.dm, with each hit mob added to said list.

Knockdown isn't granted to turn mobs horizontal, with prime examples being chairbuckling and stun immunity (while the latter will merely end with the victim being nuggeted/killed, the former can lag the game and crash clients).

Reply:
EDIT: was pointed out this specific object will end up embedding in most of these situations and not crash clients.
Still, I believe this is a solid suggestion to avoid these theorical issues, as well as instacrits and nuggetting.